### DISCUSSION TITLE
Localizations for certain components

### DISCUSSION THREAD

**Discussion by RoyRao2333** [saadeghi/daisyui#2361] [Ideas / feature requests]
Like file input, we can't change the text in it:

<img width="1331" alt="image" src="https://github.com/saadeghi/daisyui/assets/31413093/0254291c-bd7e-42d0-927e-8a821003b9f5">

I hope daisyUI can wrap something like `<DaisyProvider localization="zh-CN"><App /></DaisyProvider>` around the App's root.

Currently, if I change `lang` to `zh` in `index.html`, nothing will happen. Really need this for an web App to support multiple languages.

**Comment by saadeghi** [comment]
1. `Choose File` text is not from daisyUI. As [you can see here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file), it's the default text for the HTML element and it's controlled by browser.
2. daisyUI is just CSS and native HTML elements. It's not a JS component. There's no "provider", etc.
3. Unfortunately there's no way to change the button text inside `<input type="file">` in HTML, But you can hide the default button and use a custom button like this: https://play.tailwindcss.com/6r3k3rWvhL or you can put another text on it like this: https://play.tailwindcss.com/tCjo9RXeol

**Reply by RoyRao2333** [reply]
Wow, thanks for the solution!
